dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,345 @@
+
+Additional consensus subsidy checks:
+* The number treasurybase TxOut scripts shall be 2.
+* The version of TxOut[0] script shall be 0.
+* The length of TxOut[0] script shall be 1 and the script shall be OP_TADD.
+* The treasury subsidy shall be calculated based on the hight and verified to be correct in the transaction. One notable difference is that the treasury subsidy is NOT divided by he number of votes on the block. This is the only difference from the original coinbase subsidy calculation.
+
+Changes to coinbase:
+* Coinbase shall no longer carry the treasury subsidy transaction at TxOut[0].
+* Coinbase transaction version shall be 3.
+* The treasury subsidy value is subtracted from TxIn[0] value.
+
+===Credit treasury===
+
+A user can send utxos to the treasury for a variety of reasons such as
+donations, overpaid contractor, moving old treasury balance to new treasury
+account etc.
+
+A treasury add transaction is identified by the following characteristics:
+* The wire message transaction version shall be 3.
+* There shall be at least one TxIn script.
+* There shall be one or two TxOut scripts.
+* All TxOut scripts shall be version 0.
+* TxOut[0] script must be OP_TADD
+* TxOut[1] is optional if there is no change. If there is change it shall be a default OP_SSTXCHANGE script.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0..N]||Default utxo scripts||Normal utxo scripts as you would find on normal transactions
+|-
+|TxOut[0]||[0]OP_TADD||Value shall be what is credited to the treasury
+|-
+|TxOut[1]||[0]OP_SSTXCHANGE||Optional if there is no change.
+|}
+
+Additional consensus enforced characteristics:
+* A TAdd transaction shall not have a zero value for change.
+* There shall be no more than twenty TAdd transactions per block.
+
+===Debit treasury===
+
+The TSpend transactions are rather complex and some additional definitions need
+to be made prior to explaining the format.
+
+Required chain configuration parameters:
+* Treasury Vote Interval, or TVI, determines in which blocks a TSpend may occur. A block is considered on a TVI when the block height modulo TVI is equal to zero.
+* Treasury Vote Interval Multiplier, or TVIM, dictates the total duration (total duration = TVI * TVIM) where votes for or against the expenditure are considered valid.
+* Treasury Vote Quorum Multiplier and Divisor, TVQM and TVQD respectively, are used to calculate the required number of votes to reach quorum.
+* Treasury Vote Required Multiplier and Divisor, TVRM and TVRD respectively, are used to calculate the required number of Yes votes.
+* Treasury Expenditure Policy, or TEP, determines the overall duration where all treasury expenditures are summed and averaged.
+* Treasury Expenditure Window, or TEW, is used to calculate the window that is used to verify that the sum of all TSpends does not exceed ~150% of the average treasury spending.
```

Comment:
I am ok with rewording this however it was decided to base the expenditure limit to be on the average of past expenditures instead of incoming. Incoming is a decaying function and within a year or two this will be an issue. Thus it was decided to use average outgoing instead"